internal/utils/test/fixtures.go

```diff
@@ -0,0 +1,44 @@
+package test
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhatinsights/payload-tracker-go/internal/config"
+)
+
+func WithDatabase() func() *gorm.DB {
+	var db *gorm.DB
+
+	BeforeEach(func() {
+		var err error
+		cfg := config.Get()
+		var (
+			user     = cfg.DatabaseConfig.DBUser
+			password = cfg.DatabaseConfig.DBPassword
+			dbname   = cfg.DatabaseConfig.DBName
+			host     = cfg.DatabaseConfig.DBHost
+			port     = cfg.DatabaseConfig.DBPort
+		)
+
+		dsn := fmt.Sprintf("user=%s password=%s dbname=%s host=%s port=%s sslmode=disable", user, password, dbname, host, port)
+
+		db, err = gorm.Open(postgres.Open(dsn), &gorm.Config{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if sqlConnection, err := db.DB(); err != nil {
```

Comment:
should we log an error if `err` is not nil? So if the connection doesn't close and that causes something to blow up we have a log of that to confirm? Not sure how logging in tests work or if that's easily do-able tho

Reply:
I think it should spit out some tracebacks if it runs into errors in this case. Extra logging here might be a bit of an overkill, not sure though. :thinking:

Reply:
kk, yeah I think it would be overkill as well if we already will see some trace logs :+1:

Reply:
Cool! :100: